airbyte_cdk/sources/declarative/requesters/http_job_repository.py

```diff
@@ -273,24 +273,64 @@ def _clean_up_job(self, job_id: str) -> None:
         del self._create_job_response_by_id[job_id]
         del self._polling_job_response_by_id[job_id]
 
+    def _get_creation_response_interpolation_context(self, job: AsyncJob) -> Dict[str, Any]:
+        """
+        Returns the interpolation context for the creation response.
+
+        Args:
+            job (AsyncJob): The job for which to get the creation response interpolation context.
+
+        Returns:
+            Dict[str, Any]: The interpolation context as a dictionary.
+        """
+        creation_response_context = dict(self._create_job_response_by_id[job.api_job_id()].json())
+        creation_response_context["headers"] = self._create_job_response_by_id[
```

Comment:
Can we check if `headers` or `request` is already in `creation_response_context` before writing the values below?

Reply:
Done.